clang/lib/CAS/CASOptions.cpp

```diff
@@ -24,20 +25,33 @@ CASOptions::getOrCreateDatabases(DiagnosticsEngine &Diags,
   if (Cache.Config.IsFrozen)
     return {Cache.CAS, Cache.AC};
 
-  initCache(Diags);
+  if (auto E = initCache())
+    Diags.Report(diag::err_plugin_cas_cannot_be_initialized)
```

Comment:
`initCache()` previously could emit either `diag::err_plugin_cas_cannot_be_initialized` or `diag::err_builtin_cas_cannot_be_initialized` but after these changes only `diag::err_plugin_cas_cannot_be_initialized` is emitted.

Here's what I'd suggest:

* Replace both `err_builtin_cas_cannot_be_initialized` and `err_plugin_cas_cannot_be_initialized` with one diagnostic, like
```
def err_cas_cannot_be_initialized : Error<"CAS cannot be initialized: %1">,
  DefaultFatal;
```
* Create specific `StringError`s for `initCache()` to return, to have an error that distinguishes between builtin or plugin with the appropriate information.
* Also it would be good to remove `err_builtin_actioncache_cannot_be_initialized` since it's not used anymore.

Reply:
I didn't realize we have two different diagnostics. I merge them into one. I also check that when the error is related to path from command-line, it is already part of the `llvm::Error` returned so there is no need to print it again.

Tweak the diagnostics a bit and let me know how it looks now.